mempool/src/shared_mempool/coordinator.rs

```diff
@@ -215,13 +216,18 @@ async fn handle_event<V>(
                         counters::PEER_BROADCAST_EVENT_LABEL,
                         counters::START_LABEL,
                     );
+
+                    let transmission_state =
+                        TransmissionState::new(smp_clone.peer_manager.is_vfn(&peer));
+
                     bounded_executor
                         .spawn(tasks::process_transaction_broadcast(
                             smp_clone,
                             transactions,
                             request_id,
                             timeline_state,
                             peer,
+                            transmission_state,
```

Comment:
why not do this in process_txn_broadcast?

Reply:
seems we're can do this a bit later on and reduce extra interface code

Reply:
Yeah, looks like we could do it there.

Honestly, there's not great separation of responsibility in this code, but also a lot of separation that confuses me